pkg/node/manager/manager.go

```diff
@@ -455,10 +472,17 @@ func (m *manager) NodeUpdated(n nodeTypes.Node) {
 			dpUpdate = false
 		}
 
+		lbls := nodeLabels
+		// Add the CIDR labels for this node, if we allow selecting nodes by CIDR
+		if option.Config.PolicyCIDRSelectsNodes {
+			lbls = labels.NewFrom(nodeLabels)
+			lbls.MergeLabels(cidrlabels.GetCIDRLabels(prefix))
+		}
```

Comment:
Why is this necessary if `nodeIdentityLabels()` already encodes CIDR labels for each node's IP

Reply:
To minimize the diff to `resolveIdentity()`, I only changed it to _stop removing_ CIDR labels. However, we still need to add them somewhere. This seems like the most reasonable place -- the code that sets labels on a node.

Reply:
I think the PR as-is is doing a pretty reasonable job on this front. I'm not the biggest fan of spreading the logic to handle this `option.Config.PolicyCIDRSelectsNodes` check across node manager, ipcache core (ie `resolveIdentity()`), and the identity allocator.. but I don't have a particularly concrete view on how we could better abstract this to handle the various cases.